app/src/main/java/com/example/taskyy/data/remote/Workers/AgendaItemWorker.kt

```diff
@@ -0,0 +1,76 @@
+package com.example.taskyy.data.remote.Workers
+
+import android.content.Context
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.example.taskyy.data.local.room_database.TaskyyDatabase
+import com.example.taskyy.data.local.room_entity.agenda_entities.PendingReminderRetryEntity
+import com.example.taskyy.data.remote.TaskyyApi
+import com.example.taskyy.data.remote.data_transfer_objects.ReminderDTO
+import com.example.taskyy.domain.error.Result
+import com.example.taskyy.ui.enums.AgendaItemAction
+import retrofit2.HttpException
+
+class AgendaItemWorker(
+    private val appContext: Context,
+    private val params: WorkerParameters,
+    private val taskyyDatabase: TaskyyDatabase,
+    private val taskyyApi: TaskyyApi
+) : CoroutineWorker(appContext, params) {
+
+    override suspend fun doWork(): Result {
+        return checkReminderRetryTable()
+        //checkTaskRetryTable()
+        //checkEventRetryTable()
+    }
+
+    private fun checkEventRetryTable() {
+    }
+
+    private fun checkTaskRetryTable() {
+    }
+
+    private suspend fun checkReminderRetryTable(): Result {
+        val reminderList = taskyyDatabase.pendingReminderRetryDao().getPendingReminderRetries()
+        if (reminderList.isNotEmpty()) {
+            for (reminder in reminderList) {
+                when (reminder.action) {
+                    AgendaItemAction.CREATE -> {
+                        try {
+                            taskyyApi.createReminder(reminder.toReminderDTO())
+                        } catch (e: HttpException) {
+                            return Result.retry()
```

Comment:
Here, I'd have a bit more fine grained response handling. For example, if you get 401, you can be sure that the next call will fail as well, so there's no point in retrying. However, if you get 500 (server error), it could be a temporary thing which could be fixed by just retrying the request. So it really depends on the status code what you want to do there

Reply:
So if in the event there is 401 error, would I just return Result.failure() and return Result.retry() if the a 500 error?

Reply:
Correct, those were just examples though. You'll need to decide this for all common status codes that could be thrown